sv.c

```diff
@@ -4771,6 +4776,272 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
+PERL_STATIC_INLINE SV *
+S_sv_freshcopy_flags(pTHX_ SV* ssv, const I32 flags)
+{
+    /* Note: If a PVIV/PVNV/PVMG is only IOK, NOK, ROK, it is _mostly_
+     * possible to create just a headless SV to store that value.
+     * Some parts of core (Perl_amagic_call in gv.c specifically) do
+     * assume - and possibly CPAN might - that SvTYPE(dsv) == SvTYPE(ssv)
+     * though, which is why the code below does not try that type
+     * simplification. Perhaps this might be worth revisiting in the future.
+     * -- April 2025. */
+    assert(ssv);
+    svtype stype = SvTYPE(ssv);
+
+#if NVSIZE <= IVSIZE
+    ASSUME(stype > SVt_NV);
+#else
+    ASSUME(stype > SVt_IV);
+#endif
+    ASSUME(stype <= SVt_PVMG);
+    SV *dsv = newSV_type(stype);
+    U32 sflags   = SvFLAGS(ssv);
+
+    ASSUME(!(flags & SV_GMAGIC)); /* Caller should have sorted this. */
+    /* Note: Perl_sv_setsv_flags essentially ignores magic, except for
+     * taint and vstring magic. */
+
+    /* The switch cases are in the order that gcov of make/make test suggests. */
+    switch(sflags & (SVp_IOK|SVp_NOK|SVf_ROK|SVp_POK|SVf_FAKE|SVppv_STATIC)) {
+        case SVp_IOK:
+            SvIV_set(dsv, SvIVX(ssv));
+            SvFLAGS(dsv) |= sflags & (SVf_IOK|SVp_IOK|SVf_IVisUV);
+            break;
+        case SVp_POK:
+            if (UNLIKELY(SvTYPE(ssv) == SVt_INVLIST)) {
+               invlist_clone(ssv, dsv);
+               return dsv;
+            }
+            S_sv_freshcopy_POK(aTHX_ dsv, ssv, flags);
+        {
+            const char *vstr_pv;
+            STRLEN vstr_len;
+            if ((vstr_pv = SvVSTRING(ssv, vstr_len))) {
+                sv_magic(dsv, NULL, PERL_MAGIC_vstring, vstr_pv, vstr_len);
+                SvRMAGICAL_on(dsv);
+            }
+        }
+            break;
+        case SVp_POK|SVp_IOK:
+            S_sv_freshcopy_POK(aTHX_ dsv, ssv, flags);
+        {
+            const char *vstr_pv;
+            STRLEN vstr_len;
+            if ((vstr_pv = SvVSTRING(ssv, vstr_len))) {
+                sv_magic(dsv, NULL, PERL_MAGIC_vstring, vstr_pv, vstr_len);
+                SvRMAGICAL_on(dsv);
+            }
+        }
+            SvIV_set(dsv, SvIVX(ssv));
+            SvFLAGS(dsv) |= (sflags & (SVf_IOK|SVp_IOK|SVf_IVisUV));
+            if ((sflags & SVf_IOK) && !(sflags & SVf_POK)) {
+                /* Source was SVf_IOK|SVp_IOK|SVp_POK but not SVf_POK, meaning
+                   a value set as an integer and later stringified. So mark
+                   destination the same: */
+                SvFLAGS(dsv) &= ~SVf_POK;
+            }
+            break;
+        case SVp_POK|SVp_IOK|SVp_NOK|SVppv_STATIC: /* e.g. PL_sv_yes, PL_sv_no */
+            assert(SvTYPE(ssv) != SVt_INVLIST);
+
+            ASSUME(!SvPVX_const(dsv));
+            ASSUME(!SvOK(dsv));
+            ASSUME(!(SvFLAGS(dsv) & SVf_BREAK));
+{
+            const STRLEN cur = SvCUR(ssv);
+
+            (void)SvPOK_only(dsv);
```

Comment:
Why would there be any ROK+public 3, FAKE + private 3, SV head flag bits, be turned on, in a fresh SV* head of any body type?

Reply:
I don't understand the question, sorry. Please could you restate it?

At present, this is a case that handles  `PL_sv_yes` and `PL_sv_no` as the source SV. (This seemed to crop up pretty common according to gcov of test harness runs.) I left it somewhat generic in case other `SVppv_STATIC` SVs are added in the future.

Reply:
I think he means `SvPOK_on(dsv)` is sufficient since none of the flags in a fresh `dsv` (except maybe the POK flags) should be on, so there's no need for `SvPOK_only(dsv)` to clear those other flags.

Reply:
Oh yeah, good point. Thanks, I'll change that.